app/assets/javascripts/compute_resource.js

```diff
@@ -249,12 +249,17 @@ function vsphereGetResourcePools(item) {
     data: data,
     complete: function() { $(item).indicator_hide();},
     success: function(request) {
-      $('#compute_attribute_vm_attrs_resource_pool').select2('destroy').empty();
+      // The ids of the selectboxes are different for compute profiles and the create host page
+      selectbox = $('#compute_attribute_vm_attrs_resource_pool')
+      if (selectbox.length === 0) {
+        selectbox = $('#host_compute_attributes_resource_pool')
```

Comment:
A technique we've used elsewhere is to use `*=`, e.g. `$('*[id*=resource_pool]')` to match either, it might save you checking both with full IDs.